library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1816,44 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+   #[kani::proof_for_contract(NonNull::byte_sub)]
+    pub fn non_null_check_byte_sub() {
+    // Initializing array of size 10000
+    const ARR_SIZE: usize = 100000;
+    let arr: [i32; ARR_SIZE] = kani::any();
+    //Randomizing start pointer within array
+    let offset = kani::any_where(|x| *x <= ARR_SIZE);
+    // Create a NonNull pointer to the end of the array
+    let raw_ptr: *mut i32 = unsafe { arr.as_ptr().add(offset) as *mut i32 };
+    let ptr = unsafe { NonNull::new(raw_ptr).unwrap() };
+    // Choose an arbitrary count to subtract
+    let count: usize = kani::any();
+    // Ensure that the subtraction doesnt go out of bounds of array
+     kani::assume(count < ARR_SIZE - offset);
```

Comment:
This assume is imprecise because `byte_sub` subtracts in bytes, not the size of the array element. Therefore we should not use ARR_SIZE here but rather a value that represents the total bytes of this initialized array since `offset` is counted in bytes as well.

Reply:
I used the pointer generator and updated my harness